firefox-ios/Providers/RustSyncManager.swift

```diff
@@ -299,18 +299,19 @@ public class RustSyncManager: NSObject, SyncManager {
 
     public func getEngineEnablementChangesForAccount() -> [String: Bool] {
         var engineEnablements: [String: Bool] = [:]
+
+        let engines = self.creditCardAutofillEnabled ?
+            syncManagerAPI.rustTogglableEngines :
+            syncManagerAPI.rustTogglableEngines.filter({ $0 != RustSyncManagerAPI.TogglableEngine.creditcards })
```

Comment:
This check is not needed anymore. This was because we controlled cc autofill via a nimbus experiment and it's now always [true](https://github.com/mozilla-mobile/firefox-ios/blob/add3200f6d5b97cf4b52ebf8a1415e3fc686e265/firefox-ios/nimbus-features/autofillFeature.yaml#L1). ( [This](https://github.com/mozilla-mobile/firefox-ios/blob/add3200f6d5b97cf4b52ebf8a1415e3fc686e265/firefox-ios/Client/Application/AppDelegate.swift#L19-L29) where we set the value ).

```suggestion
        let engines = syncManagerAPI.rustTogglableEngines
```